### PR TITLE
feat: Registration

### PR DESCRIPTION
**Pull request by bluntbrain**

### CODE REVIEW

File: src/screens/UserOnboardingAndAuth/Registration.tsx [hunk 1]
```diff
@@ -1,14 +1,110 @@
...
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: {errors},
+  } = useForm();
+
+  const {on_register_click} = useContext(AuthContext);
+
+  React.useEffect(() => {
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/Registration.tsx:44]: import useEffect above than use it

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 
 export const useOnboardingAndAuth = () => {
   const [phone_number, set_phone_number] = useState<string>('');
   const [confirm, set_confirm] = useState<any>(null);
   const [code, set_code] = useState<string>('');
+  const [show_registration, set_show_registration] = useState<boolean>(false);
+
+  const navigation = useNavigation();
 
   async function sign_in_with_phone_number() {
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:17]: use arrow functions

File: src/screens/UserOnboardingAndAuth/Registration.tsx (hunk 1 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/Registration.tsx:44]: resolved

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx (hunk 2 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:17]: resolved

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 3]
```diff
@@ -26,26 +32,43 @@ export const useOnboardingAndAuth = () => {
     const confirmation = await auth().signInWithPhoneNumber(valid_phone_number);
     set_confirm(confirmation);
   }
+
   async function confirm_code() {
```
[the remark this answers is not in the document]
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:36]: resolved

File: src/screens/UserOnboardingAndAuth/Registration.tsx [hunk 4]
```diff
@@ -1,14 +1,110 @@
...
+        placeholder="Enter License Number ( optional )"
+      />
+
+      <Text style={styles.input_title}>Owner Name</Text>
+      <TextInput
+        style={styles.input}
+        onChangeText={text => setValue('owner_name', text)}
+        placeholder="Enter Gym Owner Name"
+      />
+      {errors.owner_name && (
```
Comment (shreyanshgoel9) [inline src/screens/UserOnboardingAndAuth/Registration.tsx:79]: use isEmpty otherwise in case of undefined it will print false and break the code

**Review by shreyanshgoel9 — CHANGES_REQUESTED** [review]

File: src/screens/UserOnboardingAndAuth/Registration.tsx (hunk 4 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/Registration.tsx]: resolved

**Pull request merged**